lib/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,145 @@
+#include "accelerometer.hpp"
+
+namespace hyped::sensors {
+
+Accelerometer::Accelerometer(std::uint8_t channel, io::HardwareI2c &i2c, core::ILogger &logger)
+    : channel_(channel),
+      logger_(logger),
+      i2c_(i2c)
+{
+}
+
+Accelerometer::~Accelerometer()
+{
+}
+
+std::uint8_t Accelerometer::getChannel()
+{
+  return channel_;
+}
+
+std::optional<std::int16_t> Accelerometer::getRawAcceleration(Axis axis)
+{
+  // based on the axis asked, choose the correct register address to read from
+  std::uint8_t low_byte_address, high_byte_address;
+  switch (axis) {
+    case Axis::x:
+      low_byte_address  = kXOutL;
+      high_byte_address = kXOutH;
+      break;
+    case Axis::y:
+      low_byte_address  = kYOutL;
+      high_byte_address = kYOutH;
+      break;
+    case Axis::z:
+      low_byte_address  = kZOutL;
+      high_byte_address = kZOutH;
+      break;
+  }
+
+  const auto low_byte = i2c_.readByte(kDeviceAddress, low_byte_address);
+  if (!low_byte) {
+    logger_.log(core::LogLevel::kFatal,
+                "Failed to read the low byte for acceleration along the %s",
+                AxisStrings[axis]);
+    return std::nullopt;
+  }
+
+  const auto high_byte = i2c_.readByte(kDeviceAddress, high_byte_address);
+  if (!high_byte) {
+    logger_.log(core::LogLevel::kFatal,
+                "Failed to read the low byte for acceleration along the %s",
+                AxisStrings[axis]);
+    return std::nullopt;
+  }
+
+  const auto raw_acceleration
+    = static_cast<std::int16_t>((high_byte.value() << 8) | low_byte.value());
+
+  return raw_acceleration;
+}
+
+std::int16_t Accelerometer::getAccelerationFromRaw(std::int16_t rawAcc)
+{
+  // these values come from the data sheet. Don't chance them.
+  std::int16_t acceleration
+    = static_cast<std::int16_t>((static_cast<int32_t>(rawAcc) * 488) / 1000);
+
+  return acceleration;
+}
+
+// todolater: check whether the range of values the accelerometer is correct, cause now the noise is
+std::optional<core::RawAccelerationData> Accelerometer::read()
+{
+  // check to see if the values are ready to be read
+  auto data_ready = i2c_.readByte(kDeviceAddress, kDataReady);
+  if (!data_ready) {
+    logger_.log(core::LogLevel::kFatal, "acceleration data could not be read");
+    return std::nullopt;
+  }
+  if (data_ready.value() % 2 == 0) {
+    logger_.log(core::LogLevel::kInfo, "acceleration data not ready yet to be read");
+    return std::nullopt;
+  }
```

Comment:
does it make a difference to the caller which of these events happen? in the first case it is `kFatal` while the second is `kInfo` which makes it seem like it does

if it does then the return values should differ